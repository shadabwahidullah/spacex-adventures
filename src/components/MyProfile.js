import React from 'react';
import { useSelector } from 'react-redux';

const MyProfile = () => {
  const rockets = useSelector((state) => state.rocketReducer);
  const missions = useSelector((state) => state.missionsReducer);

  return (
    <div className="container row">
      <div className="col-md-6 pt-2">
        <h2>My Missions</h2>
        {missions.missions.map((mission) => {
          const { name, id, reserved } = mission;
          if (reserved) {
            return (
              <h6 className="border outline-secondary m-0 p-3" key={id}>
                {name}
              </h6>
            );
          }
          return null;
        })}
      </div>
      <div className="col-md-6 pt-2">
        <h2>My Rockets</h2>
        {rockets.Rockets.map((rocket) => {
          const { rocket_name: rocketName, id, reserved } = rocket;
          if (reserved) {
            return (
              <h6 className="border outline-secondary m-0 p-3" key={id}>
                {rocketName}
              </h6>
            );
          }
          return null;
        })}
      </div>
    </div>
  );
};

export default MyProfile;
