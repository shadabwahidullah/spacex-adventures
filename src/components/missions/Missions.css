.desc {
  width: 60%;
}

.not-member,
.active-member {
  font-weight: bold;
  background-color: rgb(104, 104, 104);
  color: white;
  border: none;
  border-radius: 0.5em;
  font-size: 0.75rem;
  padding: 0.25em 1em;
}

.join-mission,
.leave-mission {
  font-weight: bold;
  color: rgb(104, 104, 104);
  background-color: transparent;
  border: 1px solid rgb(104, 104, 104);
  border-radius: 0.5em;
  font-size: 0.75rem;
  padding: 0.25em 1em;
}

.active-member {
  background-color: rgb(46, 174, 161);
}

.leave-mission {
  border: 1px solid rgb(252, 71, 71);
  color: rgb(252, 71, 71);
}

.btn:hover {
  background-color: rgb(202, 202, 202);
}
