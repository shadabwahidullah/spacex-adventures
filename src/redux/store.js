import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import rocketReducer from './RocketsReducer';
import missionsReducer from './missions/missions';

const reducer = combineReducers({
  // reducers come here
  rocketReducer,
  missionsReducer,
});

export default createStore(reducer, applyMiddleware(logger, thunk));
