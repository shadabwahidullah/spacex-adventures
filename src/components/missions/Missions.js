import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import './Missions.css';
import {
  fetchMissions,
  joinMission,
  leaveMission,
} from '../../redux/missions/missions';

let flag = true;
const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missionsReducer);

  useEffect(() => {
    if (flag) {
      dispatch(fetchMissions());
      flag = !flag;
    }
  }, []);

  return (
    <section>
      <table className="table table-striped">
        <thead>
          <tr>
            <th scope="col">Missions</th>
            <th scope="col">Description</th>
            <th scope="col">Status</th>
            <th scope="col">&nbsp;</th>
          </tr>
        </thead>
        <tbody>
          {missions.missions.map((mission) => (
            <tr key={mission.id}>
              <td>{mission.name}</td>
              <td className="desc">{mission.description}</td>
              {mission.reserved ? (
                <>
                  <td>
                    <button type="button" className="btn btn-success">
                      Active Member
                    </button>
                  </td>
                  <td>
                    <button
                      type="button"
                      className="btn btn-danger"
                      onClick={() => dispatch(leaveMission(mission.id))}
                    >
                      Leave Mission
                    </button>
                  </td>
                </>
              ) : (
                <>
                  <td>
                    <button type="button" className="btn btn-secondary">
                      Not a Member
                    </button>
                  </td>
                  <td>
                    <button
                      type="button"
                      className="btn btn-primary"
                      onClick={() => dispatch(joinMission(mission.id))}
                    >
                      Join Mission
                    </button>
                  </td>
                </>
              )}
            </tr>
          ))}
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
