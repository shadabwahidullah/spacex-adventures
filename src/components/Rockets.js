import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Rocket from './Rocket';
import { fetchRocketsCreator } from '../redux/RocketsReducer';

const Rockets = () => {
  const rockets = useSelector((state) => state.Rockets);
  console.log(rockets);
  const dispatch = useDispatch();
  useEffect(() => {
    fetchRocketsCreator(dispatch);
  }, []);
  return (
    <div>
      <Rocket />
    </div>
  );
};

export default Rockets;
