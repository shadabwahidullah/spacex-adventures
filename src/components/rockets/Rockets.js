import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Rocket from './Rocket';
import { fetchRocketsCreator } from '../../redux/RocketsReducer';

let flag = true;

const Rockets = () => {
  const rockets = useSelector((state) => state.rocketReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    if (flag) {
      fetchRocketsCreator(dispatch);
      flag = !flag;
    }
  }, []);
  return (
    <div className="rockets-list">
      {rockets.Rockets.map((rocket) => {
        const { id } = rocket;
        return <Rocket key={id} rocket={rocket} />;
      })}
    </div>
  );
};

export default Rockets;
