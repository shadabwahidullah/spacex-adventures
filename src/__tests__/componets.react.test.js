import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../components/rockets/Rockets';
import store from '../redux/store';
import Missions from '../components/missions/Missions';
import MyProfile from '../components/MyProfile';

describe('Component tests', () => {
  test('Rockets snapshot test', () => {
    const rockets = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    const tree = rockets.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('Missions snapshot test', () => {
    const rockets = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    const tree = rockets.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('My profile snapshot test', () => {
    const rockets = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );

    const tree = rockets.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
