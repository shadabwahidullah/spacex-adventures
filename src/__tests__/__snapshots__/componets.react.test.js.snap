// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="rockets-list"
/>
`;

exports[`Component tests Missions snapshot test 1`] = `
<section>
  <table
    className="table table-striped"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Missions
        </th>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Status
        </th>
        <th
          scope="col"
        >
          Â 
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;

exports[`Component tests My profile snapshot test 1`] = `
<div
  className="container row"
>
  <div
    className="col-md-6 pt-2"
  >
    <h2>
      My Missions
    </h2>
  </div>
  <div
    className="col-md-6 pt-2"
  >
    <h2>
      My Rockets
    </h2>
  </div>
</div>
`;

exports[`Component tests Rockets snapshot test 1`] = `
<div
  className="rockets-list"
/>
`;

exports[`Component tests Rockets test 1`] = `
<div
  className="rockets-list"
/>
`;
